src/_P099_XPT2046Touch.ino

```diff
@@ -0,0 +1,450 @@
+        break;
+      }
+
+    case PLUGIN_SET_DEFAULTS:
+      {
+        byte init = P099_CONFIG_STATE;
+
+        // if already configured take it from settings, else use default values
+        if(init != 1)
+        {
+          P099_CONFIG_CS_PIN      = P099_TS_CS;
+          P099_CONFIG_TRESHOLD    = P099_TS_TRESHOLD;
+          P099_CONFIG_ROTATION    = P099_TS_ROTATION;
+          P099_CONFIG_X_RES       = P099_TS_X_RES;
+          P099_CONFIG_Y_RES       = P099_TS_Y_RES;
+          P099_CONFIG_OBJECTCOUNT = P099_INIT_OBJECTCOUNT;
+
+          uint32_t lSettings = 0;
+          bitWrite(lSettings, P099_FLAGS_SEND_XY,         P099_TS_SEND_XY);
+          bitWrite(lSettings, P099_FLAGS_SEND_Z,          P099_TS_SEND_Z);
+          bitWrite(lSettings, P099_FLAGS_SEND_OBJECTNAME, P099_TS_SEND_OBJECTNAME);
+          bitWrite(lSettings, P099_FLAGS_USE_CALIBRATION, P099_TS_USE_CALIBRATION);
+          bitWrite(lSettings, P099_FLAGS_LOG_CALIBRATION, P099_TS_LOG_CALIBRATION);
+          P099_CONFIG_FLAGS  = lSettings;
+        }
+        success = true;
+        break;
+      }
+    case PLUGIN_WEBFORM_LOAD:
+      {
+        addFormSubHeader(F("Screen"));
+
+        uint16_t width_ = P099_CONFIG_X_RES;
+        if(width_ == 0)
+          width_ = P099_TS_X_RES; // default value
+        addFormNumericBox(F("Screen Width (px) (x)"), F("p099_width"), width_, 1, 65535);
+
+        uint16_t height_ = P099_CONFIG_Y_RES;
+        if(height_ == 0)
+          height_ = P099_TS_Y_RES; // default value
+        addFormNumericBox(F("Screen Height (px) (y)"), F("p099_height"), height_, 1, 65535);
+
+        byte choice2 = P099_CONFIG_ROTATION;
+        String options2[4] = { F("Normal"), F("+90°"), F("+180°"), F("+270°") };
+        int optionValues2[4] = { 2, 3, 0, 1 }; // Rotated by 180 degrees, compared to the TFT rotation!
+        addFormSelector(F("Rotation"), F("p099_rotate"), 4, options2, optionValues2, choice2);
+
+        addFormSubHeader(F("Touch configuration"));
+
+        byte treshold = P099_CONFIG_TRESHOLD;
+        addFormNumericBox(F("Touch minimum pressure"), F("p099_treshold"), treshold, 0, 255);
+
+        byte choice3 = 0;
+        bitWrite(choice3, P099_FLAGS_SEND_XY,         bitRead(P099_CONFIG_FLAGS, P099_FLAGS_SEND_XY));
+        bitWrite(choice3, P099_FLAGS_SEND_Z,          bitRead(P099_CONFIG_FLAGS, P099_FLAGS_SEND_Z));
+        bitWrite(choice3, P099_FLAGS_SEND_OBJECTNAME, bitRead(P099_CONFIG_FLAGS, P099_FLAGS_SEND_OBJECTNAME));
+        String options3[6] = { F("None"), F("X and Y"), F("X, Y and Z"), F("Objectnames only"), F("Objectnames, X and Y"), F("Objectnames, X, Y and Z")};
+        int optionValues3[6] = { 0, 1, 3, 4, 5, 7 }; // Already used as a bitmap!
+        addFormSelector(F("Events"), F("p099_events"), 6, options3, optionValues3, choice3);
+
+        {
+          P099_data_struct *P099_data = new (std::nothrow) P099_data_struct(); //static_cast<P099_data_struct *>(getPluginTaskData(event->TaskIndex));

```

Comment:
Just curious, why do you create a new data struct in the `PLUGIN_WEBFORM_LOAD` call instead of the `PLUGIN_INIT` call?
If the plugin is enabled if you load the page, then this object will be present in memory twice.